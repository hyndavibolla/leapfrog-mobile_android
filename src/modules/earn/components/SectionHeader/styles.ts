import { css } from '@emotion/native';

import { COLOR } from '_constants';

export const styles = {
  sectionHeader: css`
    padding-bottom: 12px;
    flex-direction: row;
    justify-content: space-between;
  `,
  link: css`
    color: ${COLOR.PRIMARY_BLUE};
  `
};
