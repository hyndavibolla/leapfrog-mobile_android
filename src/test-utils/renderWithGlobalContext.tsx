import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
