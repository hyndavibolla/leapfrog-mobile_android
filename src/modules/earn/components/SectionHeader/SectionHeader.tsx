import React, { memo, ComponentProps } from 'react';
import { View, TouchableHighlight } from 'react-native';

import { Title, TitleType } from '_components/Title';

import { styles } from './styles';

export interface Props {
  title: string;
  shouldShowSeeAll: boolean;
  seeAllProps?: ComponentProps<typeof TouchableHighlight>;
}

export const SectionHeader = memo(({ title, shouldShowSeeAll, seeAllProps }: Props) => {
  return (
    <View style={styles.sectionHeader}>
      <Title numberOfLines={1} ellipsizeMode="tail" type={TitleType.SECTION}>
        {title}
      </Title>
      {!shouldShowSeeAll ? null : (
        <TouchableHighlight underlayColor="transparent" {...seeAllProps}>
          <Title style={styles.link} type={TitleType.SECTION}>
            See all
          </Title>
        </TouchableHighlight>
      )}
    </View>
  );
});

export default SectionHeader;
