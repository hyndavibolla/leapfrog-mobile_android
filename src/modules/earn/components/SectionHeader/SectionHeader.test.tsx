import React from 'react';

import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import SectionHeader, { Props } from './SectionHeader';

describe('SectionHeader', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      title: 'this is a title',
      shouldShowSeeAll: true,
      seeAllProps: { onPress: jest.fn() }
    };
  });

  it('should render', () => {
    const { toJSON } = renderWithGlobalContext(<SectionHeader {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
