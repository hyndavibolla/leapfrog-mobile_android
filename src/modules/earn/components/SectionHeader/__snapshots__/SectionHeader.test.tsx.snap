// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader should render 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 12,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
      ]
    }
    type="sectionHeader"
  >
    this is a title
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Array [
          Object {
            "color": "#262C2F",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 16,
            "lineHeight": 24,
          },
          Object {
            "color": "#0066CC",
          },
        ]
      }
      type="sectionHeader"
    >
      See all
    </Text>
  </View>
</View>
`;
